hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/balancer/ContainerBalancer.java

```diff
@@ -731,6 +738,24 @@ boolean canSizeEnterTarget(DatanodeDetails target, long size) {
     return potentialTargets;
   }
 
+  /**
+   * Consults the configurations {@link ContainerBalancer#includeNodes} and
+   * {@link ContainerBalancer#excludeNodes} to check if the specified
+   * Datanode should be excluded from balancing.
+   * @param datanode DatanodeDetails to check
+   * @return true if Datanode should be excluded, else false
+   */
+  boolean shouldExcludeDatanode(DatanodeDetails datanode) {
```

Comment:
This is bad configuration. I think we should throw an error in this case.

Reply:
Should I simply throw an error or log it instead and recover by excluding the node?

Reply:
I think we should throw an IllegalArgumentsException. Otherwise it can be categorised as inconsistent behaviour.

Reply:
According to our latest discussion, in which the following points were identified:

- We can recover from this scenario by excluding the node. In the worst case, this will lead to the exclusion of a datanode that the user actually wanted to include.
- HDFS Balancer excludes the datanode if it's mentioned in both include and exclude lists.

Let's exclude the datanode from balancing in this scenario.